pinot-controller/src/test/java/org/apache/pinot/controller/api/PinotSegmentRestletResourceTest.java

```diff
@@ -195,6 +195,32 @@ private void checkCrcRequest(Map<String, SegmentMetadata> metadataTable, int exp
     assertEquals(crcMap.size(), expectedSize);
   }
 
+  @Test
+  public void checkTableSegmentMismatch()
+      throws Exception {
+    String tableName = "testTable1";
+    String createTableUrl = TEST_INSTANCE.getControllerRequestURLBuilder().forTableCreate();
+    TableConfigBuilder offlineBuilder = new TableConfigBuilder(TableType.OFFLINE);
+
+    TableConfig testTableConfig = offlineBuilder.setTableName(tableName).build();
+
+    // Create the table
+    ControllerTest.sendPostRequest(createTableUrl, testTableConfig.toJsonString());
+
+    // Rebalance table
+    ControllerTest.sendPostRequest(
```

Comment:
There is no segment in this table, so both rebalance and segment mismatch will be no-op. Let's try to generate a non-empty IS and EV with some diff, and verify the behavior